android-components/components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/store/SyncStoreSupport.kt

```diff
@@ -113,6 +113,11 @@ internal class FxaAccountObserver(
         store.dispatch(SyncAction.UpdateSyncStatus(SyncStatus.LoggedOut))
         store.dispatch(SyncAction.UpdateAccount(null))
     }
+
+    override fun onAccountDeleted() {
+        store.dispatch(SyncAction.UpdateSyncStatus(SyncStatus.LoggedOut))
```

Comment:
If you introduce a new `SyncStatus` state, then you would know when an account is deleted and you can observe this `SyncStore` to know when to do the rest of the UI changes you have in Fenix. See the above comments for context as well.

Reply:
I am still hesitant to use that approach, because of the life-cycle of fragments. events sent while the observer isn't subscribed are lost, right? we need somehow to register an account deleted status in a custom tab activity and then show snackbar in a different fragment, once the navigation happens.

Reply:
Since we're putting `Event.Account.Delete` in the same place as `Event.Account.Logout`, that would probably mean we should make it symmetrical on the `SyncStore`.